Sources/Valet/SecureEnclaveValet.swift

```diff
@@ -121,75 +121,119 @@ public final class SecureEnclaveValet: NSObject, Sendable {
     /// - Throws: An error of type `KeychainError`.
     /// - Important: Inserted data should be no larger than 4kb.
     @objc
-    public func setObject(_ object: Data, forKey key: String) throws {
-        try execute(in: lock) {
-            try SecureEnclave.setObject(object, forKey: key, options: baseKeychainQuery)
+    public func setObject(_ object: Data, forKey key: String) throws(KeychainError) {
+        lock.lock()
+        defer {
+            lock.unlock()
         }
+        return try SecureEnclave.setObject(object, forKey: key, options: baseKeychainQuery)
     }
 
+#if !os(tvOS) && !os(watchOS) && canImport(LocalAuthentication)
     /// - Parameters:
     ///   - key: A key used to retrieve the desired object from the keychain.
     ///   -  userPrompt: The prompt displayed to the user in Apple's Face ID, Touch ID, or passcode entry UI.
     /// - Returns: The data currently stored in the keychain for the provided key.
     /// - Throws: An error of type `KeychainError`.
     @objc
-    public func object(forKey key: String, withPrompt userPrompt: String) throws -> Data {
-        try execute(in: lock) {
```

Comment:
Swift 6 can't rethrow with a typed error, so we just inline the helper here and elsewhere